server/src/test/java/org/elasticsearch/search/aggregations/metrics/MaxIT.java

```diff
@@ -24,338 +24,24 @@
 import org.elasticsearch.script.Script;
 import org.elasticsearch.script.ScriptType;
 import org.elasticsearch.search.aggregations.AggregationTestScriptsPlugin;
-import org.elasticsearch.search.aggregations.Aggregator;
-import org.elasticsearch.search.aggregations.InternalAggregation;
-import org.elasticsearch.search.aggregations.bucket.filter.Filter;
-import org.elasticsearch.search.aggregations.bucket.global.Global;
-import org.elasticsearch.search.aggregations.bucket.histogram.Histogram;
-import org.elasticsearch.search.aggregations.bucket.terms.Terms;
-import org.elasticsearch.search.aggregations.BucketOrder;
+import org.elasticsearch.test.ESIntegTestCase;
 
 import java.util.Collection;
 import java.util.Collections;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
 
 import static java.util.Collections.emptyMap;
-import static org.elasticsearch.index.query.QueryBuilders.matchAllQuery;
-import static org.elasticsearch.index.query.QueryBuilders.termQuery;
-import static org.elasticsearch.search.aggregations.AggregationBuilders.count;
-import static org.elasticsearch.search.aggregations.AggregationBuilders.filter;
-import static org.elasticsearch.search.aggregations.AggregationBuilders.global;
-import static org.elasticsearch.search.aggregations.AggregationBuilders.histogram;
 import static org.elasticsearch.search.aggregations.AggregationBuilders.max;
-import static org.elasticsearch.search.aggregations.AggregationBuilders.terms;
 import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
-import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertHitCount;
 import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertSearchResponse;
 import static org.hamcrest.Matchers.equalTo;
-import static org.hamcrest.Matchers.notNullValue;
 
-public class MaxIT extends AbstractNumericTestCase {
+public class MaxIT extends ESIntegTestCase {
+
```

Comment:
I think it's ok to remove this test entirely. `testDontCacheScripts` is tested on a lot of aggregations (metrics, top_hits, terms, ...) but we don't really need an integration test for this. For instance `AbstractQueryTestCase#testToQuery` checks `QueryShardContext#isCacheable` after building the query, we could have the same kind of unit test in `MaxAggregatorTests`. Does this makes sense ?

Reply:
++ Christos and I chatted about this last week and agreed it's doable in the framework.  I'll make the change and merge (Christos was on support ramp and wasn't able to get to it before PTO).  Thanks!

Reply:
Removed integration test `MaxIT` and implemented caching checks in `MaxAggregatorTests#testCacheAggregation` (without scripts -> cached) and `MaxAggregatorTests#testDontCacheScripts` (with scripts -> not cached)